src/main/java/fi/helsinki/cs/tmc/core/communication/TmcServerCommunicationTaskFactory.java

```diff
@@ -144,18 +144,25 @@ private URI addApiCallQueryParameters(URI url) throws NotLoggedInException {
         return url;
     }
 
-    public Callable<Exercise> getAdaptiveExercise() 
-        throws OAuthSystemException, OAuthProblemException, NotLoggedInException {
-        return wrapWithNotLoggedInException(new Callable<Exercise>() {
+    public URI getSkillifierUrl(String addition) {
+        if (!addition.isEmpty()) {
+            return URI.create("http://tmc-adapt.testmycode.io/Example/default/" + addition);
+        }
+        return URI.create("http://tmc-adapt.testmycode.io/Example/default/");
```

Comment:
I dislike hard-coded URLs, consider moving this URL to the settings window in a future sprint (Consult @nygrenh)